### PR TITLE
Add support for Knob Goblin shield

### PR DESCRIPTION
**Pull request by lucasuyezu**
## TODO before opening for reviews
- [x] Add scaffolding code
- [x] Copy keymap from qmk
- [x] Flash on a nice!nano
- [x] Test on cable
- [x] Test on Bluetooth

## Board/Shield Check-list
 - [x] This board/shield is tested working on real hardware
 - [x] Definitions follow the general style of other shields/boards upstream ([Reference](https://zmk.dev/docs/development/new-shield))
 - [x] `.zmk.yml` metadata file added
 - [x] Proper Copyright + License headers added to applicable files (Generally, we stick to "The ZMK Contributors" for copyrights to help avoid churn when files get edited)
 - [ ] General consistent formatting of DeviceTree files
 - [x] Keymaps do not use deprecated key defines (Check using the [upgrader tool](https://zmk.dev/docs/codes/keymap-upgrader))
 - [x] `&pro_micro` used in favor of `&pro_micro_d/a` if applicable
 - [x] If split, no name added for the right/peripheral half
 - [x] `.conf` file has optional extra features commented out

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -41,6 +41,7 @@ jobs:
           - jian_right
           - jorne_left
           - jorne_right
+          - knob_goblin
```
Comment (Nicell) [inline .github/workflows/build.yml:44]: Please don't edit this file, we're changing how we do builds very soon!

File: app/boards/arm/nice_nano/nice_nano_v2.yaml [hunk 2]
```diff
@@ -8,6 +8,7 @@ toolchain:
   - xtools
 supported:
   - adc
+  - encoders
```
Comment (Nicell) [inline app/boards/arm/nice_nano/nice_nano_v2.yaml:11]: The nice!nano itself doesn't support encoders, no need to change this. A shield that the nice!nano plugs into may support encoders though! This also is a Zephyr specific file, which doesn't know anything about these encoders.

File: app/boards/shields/knob_goblin/Kconfig.defconfig [hunk 3]
```diff
@@ -0,0 +1,45 @@
+if SHIELD_KNOB_GOBLIN
+
+config ZMK_KEYBOARD_NAME
+  default "Knob Goblin"
+
+config ZMK_USB
+  default y
+
```
Comment (Nicell) [inline app/boards/shields/knob_goblin/Kconfig.defconfig:8]: You don't need to add this configuration. The board that is plugged in will dictate whether this is needed.
```suggestion
```

File: app/boards/shields/knob_goblin/knob_goblin.overlay [hunk 4]
```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2021 The ZMK Contributors
```
Comment (Nicell) [inline app/boards/shields/knob_goblin/knob_goblin.overlay:2]:
```suggestion
 * Copyright (c) 2022 The ZMK Contributors
```

**Review by Nicell — CHANGES_REQUESTED** [review]
Thanks for this PR! Have a few comments

**Comment by lucasuyezu** [comment]
@Nicell comments addressed.

File: app/boards/shields/knob_goblin/Kconfig.defconfig [hunk 5]
```diff
@@ -0,0 +1,42 @@
+if SHIELD_KNOB_GOBLIN
```
Comment (Nicell) [inline app/boards/shields/knob_goblin/Kconfig.defconfig:1]: Could you add the copyright header to here too?
```suggestion
# Copyright (c) 2022 The ZMK Contributors
# SPDX-License-Identifier: MIT

if SHIELD_KNOB_GOBLIN
```

File: app/boards/shields/knob_goblin/knob_goblin.conf [hunk 6]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2022 The ZMK Contributors
+# SPDX-License-Identifier: MIT
+
+# Uncomment to enable Encoders
+CONFIG_EC11=y
+CONFIG_EC11_TRIGGER_GLOBAL_THREAD=y
+
+# Uncomment the following line to enable the Knob Goblin OLED Display
+CONFIG_ZMK_DISPLAY=y
```
Comment (Nicell) [inline app/boards/shields/knob_goblin/knob_goblin.conf:9]: These should probably be commented out by default, unless every knob goblin has these

**Review by Nicell — CHANGES_REQUESTED** [review]

File: app/boards/shields/knob_goblin/knob_goblin.conf (hunk 6 above)
Comment (lucasuyezu) [reply app/boards/shields/knob_goblin/knob_goblin.conf:9]: Every knob goblin ships with those, but the user may choose to not solder the screen for example. What should I do in this case?

Comment (Nicell) [reply app/boards/shields/knob_goblin/knob_goblin.conf:9]: In this case, I think the decision should be up to you on whether they should be turned on by default. I would say it seems sane either way.

Comment (lucasuyezu) [reply app/boards/shields/knob_goblin/knob_goblin.conf:9]: In that case I'd rather leave it like this, that way the default route to install ZMK would work with zero code changes.

**Review by Nicell — APPROVED** [review]
Thanks!

**Pull request merged**